pkg/oc/cli/admin/release/new.go

```diff
@@ -1090,11 +1113,13 @@ func (o *NewOptions) write(r io.Reader, is *imageapi.ImageStream, now time.Time)
 			}
 		}
 
+		verifier := imagemanifest.NewVerifier()
 		options := imageappend.NewAppendImageOptions(genericclioptions.IOStreams{Out: o.Out, ErrOut: o.ErrOut})
-		options.RegistryConfig = o.RegistryConfig
+		options.SecurityOptions = o.SecurityOptions
 		options.DryRun = o.DryRun
 		options.From = toImageBase
-		options.ConfigurationCallback = func(dgst digest.Digest, config *docker10.DockerImageConfig) error {
+		options.ConfigurationCallback = func(dgst, contentDigest digest.Digest, config *docker10.DockerImageConfig) error {
+			verifier.Verify(dgst, contentDigest)
```

Comment:
Why do we need Verifier? Can we just return the error?
```
if dgst != contentDigest {
    err := fmt.Errorf("the base image failed content verification and may have been tampered with")
    if !o.SecurityOptions.SkipVerification {
        return err
    }
    fmt.Fprintf(o.ErrOut, "warning: %v\n", err) // klog?
}
```

Reply:
For many of these they can run in parallel, I'd rather use common code.  I think over time I might move the logic and message into verifier but it's not clear today it cleans up the code enough.  I don't want to have different checks everywhere.

We don't use klog for human facing messages in cli commands (since it prints output not geared for humans)